libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -281,6 +281,31 @@ void AP_Scripting::handle_mission_command(const AP_Mission::Mission_Command& cmd
     mission_data->push(cmd);
 }
 
+bool AP_Scripting::arming_checks(size_t buflen, char *buffer) const
+{
+    if (!enabled() || ((_debug_options.get() & uint8_t(lua_scripts::DebugLevel::DISABLE_PRE_ARM)) != 0)) {
+        return true;
+    }
+
+    if (_thread_failed) {
+        hal.util->snprintf(buffer, buflen, "Scripting: %s", "failed to start");
+        return false;
+    }
+
+    if (_init_failed) {
+        hal.util->snprintf(buffer, buflen, "Scripting: %s", "out of memory");
+        return false;
+    }
+
+    const char *error_buf = lua_scripts::get_last_error_message();
```

Comment:
Note that were now accessing this from a different thread than the one that populates it. I think its harmless, the worst case would be a nonsense message of buflen. We could add a semaphore?

Reply:
> Note that were now accessing this from a different thread than the one that populates it. I think its harmless, the worst case would be a nonsense message of buflen. We could add a semaphore?

Definitely not harmless. If it has to be on a different thread then isn't a semaphore required?

Reply:
You don't always need a semaphore, were still running on a single processor, so if operations only take a single clock cycle there is no need to protect. 

In this case its a read only pointer to some place in memory, so the worst case is we read something this is not our message. We cant go on reading for ever looking for the end of the string because were constrained to a max length of `buflen`. Obviously the worst case is unlikely, you would have to be reading the message just as the scripting error handler was writing it.

So if that is out worst case I don't think we really need to protect with a semaphore. It adds flash and complexity. 

Of course I may have missed some more dangerous outcome, that is why I highlighted it with the comment.